design/workitems.go

```diff
@@ -150,4 +150,19 @@ var _ = a.Resource("workitem", func() {
 		a.Response(d.NotFound, JSONAPIErrors)
 		a.Response(d.Unauthorized, JSONAPIErrors)
 	})
+	a.Action("reorder", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.PATCH("/reorder"),
+		)
+		a.Description("reorder the work item with the given id.")
+		a.Payload(workItemSingle)
```

Comment:
That's fine, but the API is the same, you just add support for a single WI in array; Move [X] Before Z